openedx/openedx-platform
# feat: update youtube transcript fetch to allow all languages

## Description

This PR updates the transcript functions that handle the checking and importing of YouTube closed captions to video blocks. Additionally this PR add `v0` endpoints for the YouTube transcript changes so that they are accessible by the Authoring API. Previously only English closed captions could be imported from YouTube videos, even if there were multiple languages found for a video. This PR removes the check for English language codes and allows all languages to be returned. This change impact Authors. 

## Supporting information

JIRA Ticket: [TNL-11530](https://2u-internal.atlassian.net/browse/TNL-11530)
> AC:
>
> - Adding Video Block with a youtube url using the ingestion engine automatically triggers the creation/pull of a transcript files from the youtube video.
>
> - All languages possible is ideal, english is required.

## Testing instructions

1. Find a YouTube video with languages other than english
2. Copy link the YouTube video
3. Open a course
4. Edit a video block
5. Paste YouTube video in the video url field
6. Scroll to the Transcript section
7. Click "Import"
8. Reload page
9. Check that expected languages were added to Transcript section

## Deadline

None
